File: Source/Components/HMAC.cs
namespace Source.Components;

public class HMAC
{
    public string? Signature { get; set; }
    public DateTimeOffset RequestedOn { get; init; } = DateTimeOffset.UtcNow;
    public Guid Nonce { get; init; } = Guid.NewGuid();
    public string? SigningContent { get; set; }
    public MessageContent[]? MessageContent { get; set; }
}

File: Scripts/hmacManager/hmac.js
export class Hmac {
    constructor(
        signature, 
        requestedOn, 
        nonce, 
        signingContent, 
        signedHeaders
    ) {
        this.signature = signature;
        this.requestedOn = requestedOn;
        this.nonce = nonce;
        this.signingContent = signingContent;
        this.signedHeaders = signedHeaders;
    }
}